### ISSUE TITLE
OOP support

### ISSUE THREAD

**Issue by curiousinternals** [bug-quota-code/bqlib#11]
I'd like to know how feasible it would be to provide basic OOP support. It appears that sqf has the functionality necessary, but whether it's worth it is unclear.

**Comment by curiousinternals** [comment]
If this is feasible, I doubt it would be a priority. Immediately useful stuff first!

**Comment by HeavyHat** [comment]
Agreed OOP would be nice but is not necessary and may incur issues with syntax etc. If implemented it would have to be well thought out before hand.

**Comment by curiousinternals** [comment]
In designing functions to wrap built-in functionality, I was going to follow this simple convention: The first argument will always be the built-in object being acted on and the second argument will be the arguments for that function.

Example:
```sqf
[_alpha_sl, "Alpha SL"] call BQC_fnc_group_setCallsign;
```

With the function defined [here](https://github.com/bug-quota-code/bqlib/blob/dev/source/sqf/bqc/util/callMethod.sqf) and the relevant imports this becomes:
```sqf
[_alpha_sl, _setCallsign, "Alpha SL"] call _method;
```

We can't create our own objects, but it'd be nice to be able to write code as if the built-in objects actually were objects with methods in sqf. If you see any problems with this, please let me know.
